src/import/FSHImporter.ts

```diff
@@ -416,40 +416,29 @@ export class FSHImporter extends FSHVisitor {
           valueSet.description = pair.value;
         }
       });
-    componentCtx
-      .map(vsComponentCtx => this.visitVsComponent(vsComponentCtx))
-      .forEach(vsComponent => {
-        // if vsComponent is a concept component,
-        // we may be able to merge its concepts into an existing concept component.
-        if (vsComponent instanceof ValueSetConceptComponent) {
-          const matchedComponent = valueSet.components.find(existingComponent => {
-            return (
-              existingComponent instanceof ValueSetConceptComponent &&
-              vsComponent.inclusion == existingComponent.inclusion &&
-              vsComponent.from.system == existingComponent.from.system &&
-              isEqual(sortBy(vsComponent.from.valueSets), sortBy(existingComponent.from.valueSets))
-            );
-          }) as ValueSetConceptComponent;
-          if (matchedComponent) {
-            matchedComponent.concepts.push(...vsComponent.concepts);
-          } else {
-            valueSet.components.push(vsComponent);
-          }
+    vsRuleCtx.forEach(vsRule => {
+      const rule = this.visitVsRule(vsRule);
+      // if rule is a concept component,
+      // we may be able to merge its concepts into an existing concept component.
+      if (rule instanceof ValueSetConceptComponentRule) {
+        const matchedComponent = valueSet.rules.find(existingComponent => {
+          return (
+            existingComponent instanceof ValueSetConceptComponentRule &&
+            rule.inclusion == existingComponent.inclusion &&
+            rule.from.system == existingComponent.from.system &&
+            isEqual(sortBy(rule.from.valueSets), sortBy(existingComponent.from.valueSets))
+          );
+        }) as ValueSetConceptComponentRule;
+        if (matchedComponent) {
+          matchedComponent.concepts.push(...rule.concepts);
         } else {
-          valueSet.components.push(vsComponent);
+          valueSet.rules.push(rule);
         }
-      });
-    caretValueRuleCtx.forEach(caretValueRule => {
-      const rule = this.visitCaretValueRule(caretValueRule);
-      if (rule.path) {
-        logger.error(
-          'Caret rule on ValueSet cannot contain path before ^, skipping rule.',
-          rule.sourceInfo
-        );
       } else {
-        valueSet.rules.push(this.visitCaretValueRule(caretValueRule));
+        valueSet.rules.push(rule);
       }
     });
+    valueSet.rules = valueSet.rules.filter(rule => rule);
```

Comment:
What's the purpose of this filter?  Does `valueSet.rules` potentially have null entries and you're trying to filter them out?

Reply:
Yeah sometimes `visitVsRule` will return undefined if the rule is something like a `CaretValueRule` with no a path in front. But...why not just not add the rule at all? I don't know. It made sense at the time I swear.

Reply:
OK.  Then it might benefit from a comment.  And although what you have is technically correct, it might be more clear what the intent is if you changed it to:
```ts
    valueSet.rules = valueSet.rules.filter(rule => rule != null);
```

Reply:
I wasn't clear in my questioning of myself, but I meant that I am going to just not add undefined rules to the rules array. It seems simpler and more clear what is happening.